src/lotus/client.rs

```diff
@@ -288,9 +299,23 @@ impl<T: JsonRpcClient + Send + Sync> LotusClient for LotusJsonRPCClient<T> {
 
     async fn ipc_read_subnet_actor_state(
         &self,
+        subnet_id: &SubnetID,
         tip_set: Cid,
     ) -> Result<IPCReadSubnetActorStateResponse> {
-        let params = json!([GATEWAY_ACTOR_ADDRESS, [CIDMap::from(tip_set)]]);
+        let parent = subnet_id
+            .parent()
+            .ok_or_else(|| anyhow!("no parent found"))?
+            .to_string();
+        let actor = subnet_id.subnet_actor().to_string();
+        let params = json!([
+            {
+                "Parent": parent,
+                "Actor": actor
+            },
```

Comment:
Serializing `SubnetID` directly into json doesn't work due to the `serialize_tuple`, right? 

I would suggest adding a `to_json` method in `SubnetID` that does this so we don't have to do it every time, or include in the agent a `subnet_id_to_json` function to avoid having to remember how to do it like here.

Reply:
I see that we only make this serialization here, if this is the case maybe we can keep it as-is for now.

Reply:
yeah, maybe we keep it for now.